docs/api.md

```diff
@@ -3040,6 +3040,270 @@ name,label,first_name,last_name,age,favorite_color
+    "label": "John Doe",
+    "firstName": "John",
+    "lastName": "Doe",
+    "city": "New York",
+    ...
+}
+```
+
+You can provide header `X-Client-Id` to identify the client/transaction of the consumer of the API. The value is logged and can be retrieved from the [Getting Entity Details](#reference/entities/getting-entity-details) and [Listing Entity Versions](#reference/entities/listing-entity-versions) APIs in `currentVersion.source.id`
+
++ Request (application/json)
+    + Attributes
+        + projectId: `16` (number, required) - The numeric ID of the Project
+        + name: `people` (string, required) - Name of the Dataset
+
+    + Body
+
+            { 
+                "uuid": "54a405a0-53ce-4748-9788-d23a30cc3afa",
+                "label": "John Doe",
+                "firstName": "John",
+                "lastName": "Doe",
+                "city": "New York",
+                "...other properties": "...value of other properties"
+            }
+
++ Response 200 (application/json)
+    + Attributes (array[Entity Summary])
+
++ Response 200 (application/json; extended)
+    + Attributes (array[Extended Entity Summary])
+
++ Response 403 (application/json)
+    + Attributes (Error 403)
+
+## Updating an Entity [PUT /projects/{projectId}/datasets/{name}/entities/{uuid}]
+
+Use this API to completely replace the Entity. A new version of the Entity will be created with the provided values. If you want to replace just one or few properties of the Entity then please use [Updating an Entity partially](#reference/entities/updating-an-entity-partially)
+
++ Request (application/json)
+    + Attributes
+        + projectId: `16` (number, required) - The numeric ID of the Project
+        + name: `people` (string, required) - Name of the Dataset
+        + uuid: `54a405a0-53ce-4748-9788-d23a30cc3afa` (string, required) - UUID of the Entity
+
+    + Body
+
+            { 
+                "uuid": "54a405a0-53ce-4748-9788-d23a30cc3afa",
+                "label": "John Doe",
+                "firstName": "John",
+                "lastName": "Doe",
+                "city": "New York",
+                "...other properties": "...value of other properties"
+            }
+
++ Response 200 (application/json)
+    + Attributes (array[Entity Summary])
+
++ Response 200 (application/json; extended)
+    + Attributes (array[Extended Entity Summary])
+
++ Response 403 (application/json)
+    + Attributes (Error 403)
+
+## Updating an Entity partially [PATCH /projects/{projectId}/datasets/{name}/entities/{uuid}]
+
+Use this API to replace one or few properties of the Entity
+
++ Request (application/json)
+    + Attributes
+        + projectId: `16` (number, required) - The numeric ID of the Project
+        + name: `people` (string, required) - Name of the Dataset
+        + uuid: `54a405a0-53ce-4748-9788-d23a30cc3afa` (string, required) - UUID of the Entity
+
+    + Body
+
+            { 
+                "firstName": "John"
+            }
```

Comment:
I'm wondering what would happen to this endpoint and also the PUT endpoint if one day we add editable metadata to entities, something like review state. With that possibility in mind, what do you think about nesting the entity data in a property named `data` in the request body? Here's an example request body:

```json
{
  "data": {
    "firstName": "John"
  }
}
```

Another idea: what if creating a new version happened from a POST to &hellip;/entities/:uuid/versions? Then PATCH to the entity itself could be reserved only for future metadata properties.

Reply:
in that case I would like to have similar body for POST and PUT as well

Reply:
That makes sense for the endpoints to be consistent. We've struggled a little with the PUT endpoint for a project because whenever we add a property to the project, we need to change our PUT requests to include the new property. I'm worried we could run into similar issues here (though it's not often that new properties are added). Do you think we could get by with just PATCH? A user could still use PATCH to specify all the properties at once, including blanking out a property.

Reply:
let's go with the PATCH only

Reply:
How should the user update the entity label? As a top-level property next to `data` or only as a property within `data`? If the latter, I think it'd be helpful to note that in the docs.